### PR TITLE
Plans: Automatically provision premium services when signing up for a plan

### PR DESCRIPTION
**Pull request by ryelle**
When signing up for a jetpack plan, we leave the user on their own to install VaultPress, Akismet, etc. We can do this through Jetpack, automatically.

After a completed checkout, the user is redirected to a new page (currently `plugins/setup/$site`, but that can change). This checks that we can install plugins, and if so, it can start the provisioning process.

There are 3 views you can get on `plugins/setup/$site`:

- **User has a site with manage, and we can edit files**: The provisioning will start, going step-by-step through the process of installing a plugin, activating it, and making sure it succeeds. This'll happen for VaultPress, Akismet, and Polldaddy (if on Business). I'm thinking I'll use a similar approach to the steps as the checkout process (TransactionFlow), but open to suggestion.
- **User has a site with manage, but we can't edit files**: We'll show a page with instructions for installation, maybe along with their keys? Could use a designer's help with this.
- **User doesn't have manage**: We show the "Please activate manage" component. If possible, it would be nice to redirect the user back to this page once manage is activated.

If a user without a Premium or Business JP plan tries to go to the setup page, they're redirected back to plugins (or if they're a free JP user, the plans page).

### CODE REVIEW

File: client/my-sites/plugins/controller.js [hunk 1]
```diff
@@ -180,6 +217,32 @@ controller = {
 		renderPluginsBrowser( context, siteUrl );
 	},
 
+	setupPlugins: function() {
+		let selectedSite = sites.getSelectedSite();
+
+		// Not a Jetpack plan
+		if ( 0 !== selectedSite.plan.product_slug.indexOf( 'jetpack_' ) ) {
```
Comment (johnHackworth) [inline client/my-sites/plugins/controller.js]: I think I would move all this logic to `/plan-setup`. In plugins, we usually include all this checks & fallbacks (manage & permissions error views, etc) in the main component of the view we are going to show, so the controller only includes param processing & initialization of main components. This is what I mean: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins/plugin.jsx#L293  (that `/plugins/plugin.jsx` is the main component that renders a single plugin view).

File: client/my-sites/plugins/controller.js [hunk 2]
```diff
@@ -149,6 +152,40 @@ function renderPluginsBrowser( context, siteUrl ) {
...
+function renderNoManageWarning() {
+	let site = sites.getSelectedSite();
+	ReactDom.render(
+		React.createElement( JetpackManageErrorPage, {
+			site: site,
+			template: 'optInManage',
+			title: i18n.translate( 'Oh no! We can\'t automatically install your new plugins.' ),
+			section: 'plugins',
+			illustration: '/calypso/images/jetpack/jetpack-manage.svg',
+		} ),
```
Comment (johnHackworth) [inline client/my-sites/plugins/controller.js]: Does it make sense to include a feature-example (https://wpcalypso.wordpress.com/devdocs/design/feature-example) here as we do with some other JetpackManageErrorPages?

File: client/my-sites/plugins/plan-setup/style.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+			opacity: 0.7;
+		}
+	}
+}
+
+/* Only apply this to lists of 3 items */
+.plugin-plan:nth-child(1):nth-last-child(3),
+.plugin-plan:nth-child(2):nth-last-child(2),
+.plugin-plan:nth-child(3):nth-last-child(1) {
+	width: 33.333333%;
```
Comment (johnHackworth) [inline client/my-sites/plugins/plan-setup/style.scss]: I guess we can make this with flux, but I'm not good enough with it, so let's ping @MichaelArestad & @rickybanister and see if they have any proposal :D

Comment (rickybanister) [inline client/my-sites/plugins/plan-setup/style.scss]: I don't really have a sense of context of what we're trying to advise on here—is this css copied from the regular plans page? Is it new code? I might need to see screenshots or just get some steps to see this view rendered.

**Comment by johnHackworth** [comment]
I've rebased this with the last version of master, resolved the conflicts & moved `PlanSetupInstructions` initilization to inside of the main `Plan-setup` component (from `plugins/controller`)

**Comment by johnHackworth** [comment]
The test were failing in circleCI, so I've rebased this with master and now they are green again

**Comment by johnHackworth** [comment]
It worked! the single instructions file commit is merged here too now

**Comment by ryelle** [comment]
The functionality here has been added in #2788, #2944, #3908, this PR is no longer needed.

**Pull request closed without merging**